components/function-controller/pkg/controller/function/function_controller.go

```diff
@@ -367,168 +369,154 @@ func generateFunctionHash(fnCm *corev1.ConfigMap) (string, error) {
 	return fmt.Sprintf("%x", hash.Sum(nil)), nil
 }
 
-// syncFunctionBuildTemplate reconciles the current Function BuildTemplate with its
+// syncFunctionBuildTask reconciles the current Function build Task with its
 // desired state.
-func (r *ReconcileFunction) syncFunctionBuildTemplate(fn *serverlessv1alpha1.Function, ri *runtimeUtil.RuntimeInfo) error {
-	desiredBt := &buildv1alpha1.BuildTemplate{
+func (r *ReconcileFunction) syncFunctionBuildTask(fn *serverlessv1alpha1.Function, ri *runtimeUtil.RuntimeInfo) error {
+	desiredTask := &tektonv1alpha1.Task{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      buildTemplateName,
+			Name:      buildTaskName,
 			Namespace: fn.Namespace,
 		},
-		Spec: runtimeUtil.GetBuildTemplateSpec(ri),
+		Spec: *runtimeUtil.GetBuildTaskSpec(ri),
 	}
 
-	if err := controllerutil.SetControllerReference(fn, desiredBt, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(fn, desiredTask, r.scheme); err != nil {
 		return fmt.Errorf("error setting controller reference: %s", err)
 	}
 
-	currentBt, err := r.getOrCreateFunctionBuildTemplate(desiredBt)
+	currentTask, err := r.getOrCreateFunctionBuildTask(desiredTask)
 	if err != nil {
 		if err := r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError); err != nil {
 			log.Error(err, "Error setting Function status", "namespace", fn.Namespace, "name", fn.Name)
 		}
 
-		log.Error(err, "Error during sync of the Function BuildTemplate", "namespace", desiredBt.Namespace, "name", desiredBt.Name)
-
 		return err
 	}
 
-	if _, err = r.updateFunctionBuildTemplate(currentBt, desiredBt); err != nil {
+	if _, err = r.updateFunctionBuildTask(currentTask, desiredTask); err != nil {
 		if err := r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError); err != nil {
 			log.Error(err, "Error setting Function status", "namespace", fn.Namespace, "name", fn.Name)
 		}
 
-		log.Error(err, "Error during sync of the Function BuildTemplate", "namespace", desiredBt.Namespace, "name", desiredBt.Name)
+		log.Error(err, "Error during sync of the Function build Task", "namespace", desiredTask.Namespace, "name", desiredTask.Name)
```

Comment:
Why keep this errormessage but remove a similar one in 391?

Reply:
Nice catch, that error moved at line 190, this one is a leftover.
Removed.